### PR TITLE
Ameymwalimbe shift get post

### PR DESCRIPTION
**Pull request by ameymwalimbe**
In this PR, I add the backend Post and Get functionality to the shift table in #8. Michael helped me with the Post tests, and it works on Swagger. From this, drivers should be able to add their ride shifts, and any user can get all ride shifts. The shiftStart and end parameters in shift entity has been changed to type String instead of LocalTime, as I was unable to test on LocalTime objects in Swagger.

### CODE REVIEW

**Review by mike-k999 — DISMISSED** [review]
LGTM.  We will need to check at the time of the api call that the time strings match format @ru1nw

**Review by zachmiller-ucsb — DISMISSED** [review]
Format Api documentation in Swagger to make it clear what the format of post endpoint attributes need to be. For example, the days needs to be Monday, Tuesday, Wednesday, etc which was unclear. Also, the times work when it is something like 10:30 for POST, but not 10:30am or 10:30AM despite swagger saying the attribute should just be type string

File: src/main/java/edu/ucsb/cs156/gauchoride/controllers/ShiftController.java [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+    public Shift shiftByID(
+            @ApiParam(name = "id", type = "Long", value = "id number of shift to get", example = "1", required = true) @RequestParam Long id)
+            throws JsonProcessingException {
+        Shift shift = shiftRepository.findById(id)
+                .orElseThrow(() -> new EntityNotFoundException(Shift.class, id));
+        return shift;
+    }
+
+    @ApiOperation(value = "Create a new shift for the table")
+    @PreAuthorize("hasRole('ROLE_DRIVER')")
```
Comment (strongani) [inline src/main/java/edu/ucsb/cs156/gauchoride/controllers/ShiftController.java]: Maybe Admins should also be able to make a new shift.

**Review by strongani — CHANGES_REQUESTED** [review]
change tests as necessary?

**Review by strongani — APPROVED** [review]
LGTM!

**Review by pconrad — APPROVED** [review]
LGTM

**Pull request merged**